packages/graphql/src/translate/create-projection-and-params.ts

```diff
@@ -645,3 +638,41 @@ function createProjectionAndParams({
 }
 
 export default createProjectionAndParams;
+
+const getRequiredFields = ({
+    node,
+    resolveTree,
+}: {
+    node: Node;
+    resolveTree: ResolveTree;
+}): Record<string, ResolveTree> => {
+    const nodeFields = resolveTree.fieldsByTypeName[node.name];
+
+    return Array.from(
+        // Create set to remove duplicate fields required across different `@ignore` fields
+        new Set(
+            node.ignoredFields
+                // Filter any that have required fields
+                .filter((f) => f.requiredFields?.length)
+                // Filter those ignored fields that exist in selection set
+                .filter((ignoredField) => Object.values(nodeFields).find((f) => f.name === ignoredField.fieldName))
+                // Extract required fields
+                .map((f) => f.requiredFields)
+                .flat()
+        )
```

Comment:
**suggestion (non-blocking** I think we may be able to avoid some comments and nested code with a couple of extra variables and/or functions:

```suggestion
    const fieldsWithRequiredFields = node.ignoredFields.filter((f) => f.requiredFields?.length);
    const fieldsInSelectionSet = filterFieldsInSelectionSet(nodeFields, fieldsWithRequiredFields);
    const requiredFields = fieldsInSelectionSet.map((f) => f.requiredFields).flat();

    return Array.from(
        // Create set to remove duplicate fields required across different `@ignore` fields
        new Set(
            requiredFields
        )
```


```typescript
function filterFieldsInSelectionSet<T extends BaseField>(nodeFields: Record<string, ResolveTree>, fields: T[]): T[] {
    const fieldsValues = Object.values(nodeFields);
    return fields.filter((field) => fieldsValues.find((f) => f.name === field.fieldName));
}
```

This is just a suggestion on how we may do it, maybe there are better solutions

Reply:
There are other PRs that are open that can make use of utility functions on the resolve tree: https://github.com/neo4j/graphql/pull/842/commits/a9f37e2f4f964e1e3dd85016497b3e8306d86624#diff-55c04c06457213093b423b2d53edd073daf6dd676e3aca304af9f3234bdbff23